### ISSUE TITLE
default-user and default-password in client.yml not working

### ISSUE THREAD

**Issue by pinpox** [binwiederhier/ntfy#460]
I've created a file in ~/.config/ntfy/client.yml with this content
```yml
default-host: https://push.pablo.tools
default-user: pinpox
default-password: mysupersecretspassword
```
But the client seems to ingnore the values:
```
$ ntfy pub --config ~/.config/ntfy/client.yml https://push.pablo.tools/test                                                                                                                                                         
{"code":40301,"http":403,"error":"forbidden","link":"https://ntfy.sh/docs/publish/#authentication"}
```

Using the `NTFY_USER` environment variable works, so I know the crendentials are in fact correct:
```
$ export NTFY_USER=pinpox:mysupersecretspassword
$ ntfy pub test test                                                                                                                                                                                                                
{"id":"ix6JqNqNDETe","time":1667306079,"event":"message","topic":"test","message":"test"}
```

**Comment by binwiederhier** [comment]
So I looked at the code and it _should_ work. I'll try this out. :thinking:

**Comment by binwiederhier** [comment]
> Note that this leaks/publishes notifications send to a private instance with the 'ntfy cli', to the public instance at `https://ntfy.sh/*`, because the configured `default-host` is ignored.

My evil plan to collect user passwords is working :smiling_imp: -- Just kidding. You're right, I should look at this urgently.

**Comment by wunter8** [comment]
@notDavid Can you confirm notifications are being leaked? I just tested it locally, and it seems like `default-host` is being used correctly. I think only `default-user` and `default-password` are being ignored

**Comment by wunter8** [comment]
I figured out what's going on.

The [code](https://github.com/binwiederhier/ntfy/commit/9d0231ea07768edf2dbe7ffc42c88263a7abaf0e) to use `default-user` and `default-password` for `ntfy **sub**` was released in version 1.28.0.

But the [code](https://github.com/binwiederhier/ntfy/commit/85eb9160d81aef0ed4649f709d27da6e9393815c) to use `default-user` and `default-password` for `ntfy **pub**` hasn't been released yet. It's included in (unreleased) version 1.29.0.

**Comment by binwiederhier** [comment]
As per @wunter8's comment, this was fixed in #431. I'm going to release the code today.

**Comment by wunter8** [comment]
Can you confirm what version of `ntfy` you're using? (`ntfy -h | tail -n2`)

**Comment by wunter8** [comment]
What happens when you subscribe to a topic? Does it use the configured `default-host`? (`ntfy sub --trace mytopic`)

**Comment by wunter8** [comment]
Alright. Now let's check file permissions: `ls -al ~/.config/ntfy/client.yml`

**Comment by wunter8** [comment]
Oh, you're on Mac? I think the default path for `client.yml` on MacOS is `~/Library/Application Support/ntfy/client.yml`. So you could either move your config file there, or specify the path to the config file in the command: `ntfy sub --config ~/.config/ntfy/client.yml mytopic`

**Comment by binwiederhier** [comment]
(I just released the v1.29.0, so you should also try that.)

**Comment by wunter8** [comment]
I found the default path for MacOS here: https://ntfy.sh/docs/subscribe/cli/#subscribe-to-multiple-topics

But I agree we should probably update the docs elsewhere
